storage/src/store/mod.rs

```diff
@@ -0,0 +1,919 @@
+//! An unauthenticated key-value database supporting variable-sized values.
+//!
+//! # Terminology
+//!
+//! A _key_ in an unauthenticated database either has a _value_ or it doesn't. The _update_ operation
+//! gives a key a specific value whether it previously had no value or had a different value.
+//!
+//! Keys with values are called _active_. An operation is called _active_ if (1) its key is active,
+//! (2) it is a [`Operation::Set`] operation, and (3) it is the most recent operation for that key.
+//!
+//! # Lifecycle
+//!
+//! 1. **Initialization**: Create with [`Store::init`] using a [`Config`]
+//! 2. **Insertion**: Use [`Store::update`] to set a value for a given key
+//! 3. **Queries**: Use [`Store::get`] to retrieve current values
+//! 4. **Deletions**: Use [`Store::delete`] to remove a key's value
+//! 5. **Persistence**: Call [`Store::commit`] to make changes durable
+//! 6. **Cleanup**: Call [`Store::close`] to shutdown gracefully or [`Store::destroy`] to remove all data
+//!
+//! # Pruning
+//!
+//! The database prunes _inactive_ operations every time [`Store::commit`] is called. To achieve this,
+//! an _inactivity floor_ is maintained, which is the location at which all operations before are inactive.
+//! At commit-time, the inactivity floor is raised by the number of uncommitted operations plus 1 for the
+//! tailing commit op. During this process, any encountered active operations are re-applied to the tip of
+//! the log.
+//!
+//! |                               Log State                                | Inactivity Floor | Uncommitted Ops |
```

Comment:
Recommend adding a test that exactly replicates this case to ensure your counts are right.